python/cudf/cudf/io/dlpack.py

```diff
@@ -72,13 +75,27 @@ def to_dlpack(cudf_obj):
         raise ValueError("Cannot create DLPack tensor of 0 size")
 
     if isinstance(cudf_obj, (DataFrame, Series, BaseIndex)):
-        gdf_cols = cudf_obj
+        gdf = cudf_obj
     elif isinstance(cudf_obj, ColumnBase):
-        gdf_cols = cudf_obj.as_frame()
+        gdf = cudf_obj.as_frame()
     else:
         raise TypeError(
             f"Input of type {type(cudf_obj)} cannot be converted "
             "to DLPack tensor"
         )
 
-    return libdlpack.to_dlpack(gdf_cols)
+    if any(
```

Comment:
It's perhaps slightly more future proof to error based on a dtype not being present in a predefined supported set rather then checking if a dtype is in an unsupported set. That way you dont have to worry about updating it here as we add dtypes

Reply:
Yeah, also I think we can be more direct here:
```
if any(not cudf.api.types.is_numeric(col.dtype) for col in gdf._data._columns)
```

Reply:
Changed it to be more resilient with `cudf.api.types._is_non_decimal_numeric_dtype`.